catalog/resource_catalog.go

```diff
@@ -131,7 +131,20 @@ func ResourceCatalog() *schema.Resource {
 			// So if we don't update the field then the requests would be made to old Name which doesn't exists.
 			d.SetId(ci.Name)
 
-			return nil
+			if d.Get("isolation_mode") != "ISOLATED" {
+				return nil
+			}
+			// Bind the current workspace if the catalog is isolated, otherwise the read will fail
+			currentMetastoreAssignment, err := w.Metastores.Current(ctx)
+			if err != nil {
+				return err
+			}
+			_, err = w.WorkspaceBindings.Update(ctx, catalog.UpdateWorkspaceBindings{
```

Comment:
I'm not sure what API is used under the hood. Plain Update API is marked as deprecated now: https://docs.databricks.com/api/workspace/workspacebindings/update

Reply:
yes, we do have a new API now, covered in #2803 for `databricks_catalog_workspace_binding`. I've switched to the new one